src/nodes/delta/delta.jl

```diff
@@ -0,0 +1,245 @@
+export DeltaFn
+
+struct DeltaFnCallableWrapper{F} end
+
+(::Type{DeltaFnCallableWrapper{F}})(args...) where {F} = F.instance(args...)
+
+struct DeltaFn{F} end
+struct DeltaFnNode{F, N, L, M} <: AbstractFactorNode
```

Comment:
Let's discuss the naming of this node. I completely understand why it is called as it is, but then let's update the filename to `deltafn.jl`. Personally my preference would be to write `DeltaFunction` or `Delta` to prevent abbreviations similarly as in the rest of the nodes.

Also we might want to reflect on the (naming of the) `PointMass` object, as this is basically the same thing for a constrained set of functions.

Reply:
I'd suggest to just use `Delta` instead of `DeltaFn`, the "function" affix does not really add any information for me.

And I like the insight on the `PointMass`, which is essentially a `Delta` node `\delta(x - g(x))` with `g(x) = c` fixed. Although that might be for a different issue.

Reply:
@bvdmitri thoughts?